core/trino-main/src/main/java/io/trino/operator/output/PositionsAppenderPageBuilder.java

```diff
@@ -30,26 +31,40 @@ public class PositionsAppenderPageBuilder
     private static final int DEFAULT_INITIAL_EXPECTED_ENTRIES = 8;
     @VisibleForTesting
     static final int MAX_POSITION_COUNT = PageProcessor.MAX_BATCH_SIZE * 4;
+    // Maximum page size before being considered full if all RLE channels were to be converted to direct
```

Comment:
comment is misleading as `getDirectSizeInBytes` works with any appender, not only RLEs

Reply:
Updated the comment to be more explicit about the behaviors of each of the modes